From 7d70f6158cd104c5a7ffbff34fbdb605e0474188 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Mon, 03 Mar 2025 14:35:00 +0200
Subject: [PATCH] [FSW-14050] scripts: fs1: prepare bloader for comp

For palladium JGS validation, compute DIE will also need
its own bloader, as there may be some known differences from
io-die in funtiontionality.
Hence, as a preparation, this patch prepares to binary
for each, and also using different compilation flag value
in DIE_TYPE

Change-Id: I421617f184d6391ab0f1453304373ee8ce6bb123
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/app/palad_bloader/Makefile b/app/palad_bloader/Makefile
index 401b79b..a45cc0a 100644
--- a/app/palad_bloader/Makefile
+++ b/app/palad_bloader/Makefile
@@ -22,9 +22,23 @@
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 
-SFLAGS := $(CFLAGS) -D__ASSEMBLY__
+DIE_TYPE_IO = 5
+DIE_TYPE_COMP = 10
+
+CFLAGS_IO = $(CFLAGS)
+CFLAGS_IO += -DDIE_TYPE=$(DIE_TYPE_IO)
+CFLAGS_COMP += $(CFLAGS)
+CFLAGS_COMP += -DDIE_TYPE=$(DIE_TYPE_COMP)
+
+SFLAGS_IO := $(CFLAGS_IO) -D__ASSEMBLY__
+SFLAGS_COMP := $(CFLAGS_COMP) -D__ASSEMBLY__
+
+PPBOOT_LD_SCRIPT_IO := $(HL_DEV_ARCH)/ppboot.lds
+PPBOOT_LD_SCRIPT_COMP := $(HL_DEV_ARCH)/ppboot_comp.lds
 
 PALAD_BLOADER_LD_SCRIPT := $(shell ls $(HL_DEV_ARCH)/palad_bloader.lds.S)
+COMP_PALAD_BLOADER_LD_SCRIPT := $(shell ls $(HL_DEV_ARCH)/palad_bloader.lds.S)
+
 LDFLAGS := -g -nostdlib -static
 LDFLAGS += -Wl,--gc-sections
 LDLIBS := -lgcc
@@ -45,25 +59,44 @@
 PALAD_BLOADER_OBJECTS := $(PALAD_BLOADER_SRCS_C:.c=.o)
 PALAD_BLOADER_OBJECTS += $(PALAD_BLOADER_APP_S:.S=.o)
 
+COMP_PALAD_BLOADER_OBJECTS := $(PALAD_BLOADER_SRCS_C:.c=_comp.o)
+COMP_PALAD_BLOADER_OBJECTS += $(PALAD_BLOADER_APP_S:.S=.o)
+
 # TODO: add version info
 
 all: clean
-	$(MAKE) palad_bloader.bin
+	$(MAKE) palad_bloader.bin palad_bloader_comp.bin
 
 palad_bloader.bin: palad_bloader.elf
+	$(OBJCOPY) -O binary $< $@
+
+palad_bloader_comp.bin: palad_bloader_comp.elf
 	$(OBJCOPY) -O binary $< $@
 
 palad_bloader.elf: $(PALAD_BLOADER_OBJECTS) $(PALAD_BLOADER_LD_SCRIPT)
 	@$(CC) $(LDFLAGS) -T $(PALAD_BLOADER_LD_SCRIPT) -o $@ -Wl,--start-group $(PALAD_BLOADER_OBJECTS) $(LDLIBS) -Wl,--end-group
 
+palad_bloader_comp.elf: $(COMP_PALAD_BLOADER_OBJECTS) $(COMP_PALAD_BLOADER_LD_SCRIPT)
+	@$(CC) $(LDFLAGS) -T $(COMP_PALAD_BLOADER_LD_SCRIPT) -o $@ -Wl,--start-group $(COMP_PALAD_BLOADER_OBJECTS) $(LDLIBS) -Wl,--end-group
+
+
 %.lds: %.lds.S
 	@$(CC) $(SFLAGS) -E -P -nostdinc -o $@ $<
 
+%_comp.lds: %.lds.S
+	@$(CC) $(SFLAGS_COMP) -E -P -nostdinc -o $@ $<
+
 %.o: %.c
-	@$(CC) $(CFLAGS) -o $@ $<
+	@$(CC) $(CFLAGS_IO) -o $@ $<
 
 %.o: %.S
-	@$(CC) $(SFLAGS) -o $@ $<
+	@$(CC) $(SFLAGS_IO) -o $@ $<
+
+%_comp.o: %.c
+	@$(CC) $(CFLAGS_COMP) -o $@ $<
+
+%_comp.o: %.S
+	@$(CC) $(SFLAGS_COMP) -o $@ $<
 
 clean:
 	rm -f $(PALAD_BLOADER_OBJECTS)
